src/hotspot/share/logging/logConfiguration.cpp

```diff
@@ -505,6 +505,12 @@ bool LogConfiguration::parse_log_arguments(const char* outputstr,
       errstream->print_cr("Invalid output index '%s'", outputstr);
       return false;
     }
+  } else if (0 == strcmp(outputstr, StdoutLog->name())) { // stdout
+    idx = 0;
+    assert(find_output(outputstr) == idx, "sanity check");
+  } else if (0 == strcmp(outputstr, StderrLog->name())) { // stderr
+    idx = 1;
+    assert(find_output(outputstr) == idx, "sanity check");
```

Comment:
oh, I don't know there's a term for that. 

I guess I derive this from java. "stdout".equals(name) can compare name == null whereas name.equals("stdout") will throw NPE. 
 
okay. I will flip them.

Reply:
Sometimes in HotSpot you will see:
```
void *ptr = nullptr;
<some code>
if (nullptr == ptr) {
  <some code>
}
```

Some folks code variable checks that way because this:

`if (nullptr = ptr)`

will result in a compiler error whereas:

`if (ptr = nullptr)`

will not result in a compiler error. I've never heard that style called a Yoda condition before...

Reply:
I don't think I've seen it done with a function call return value though...

Reply:
I read the wiki page. It mentioned that gcc -Wall will emit a warning when we use assignment as a condition. Defensive code style like `nullptr == ptr` may have been out of fashion nowadays.